packages-exp/performance-exp/src/index.ts

```diff
@@ -17,32 +17,61 @@
 
 import '@firebase/installations';
 
-import { FirebaseApp } from '@firebase/app-types';
-import { _FirebaseNamespace } from '@firebase/app-types/private';
+import { FirebaseApp } from '@firebase/app-types-exp';
 import {
-  FirebasePerformance,
-  PerformanceSettings
+  PerformanceSettings,
+  FirebasePerformance
 } from '@firebase/performance-types-exp';
-
-import { PerformanceController } from './controllers/perf';
-import { setupApi } from './services/api_service';
-import { SettingsService } from './services/settings_service';
 import { ERROR_FACTORY, ErrorCode } from './utils/errors';
+import { setupApi } from './services/api_service';
+import { PerformanceController } from './controllers/perf';
+import { _registerComponent, _getProvider } from '@firebase/app-exp';
+import {
+  Provider,
+  InstanceFactory,
+  ComponentContainer,
+  Component,
+  ComponentType
+} from '@firebase/component';
 
 const DEFAULT_ENTRY_NAME = '[DEFAULT]';
 
 export function getPerformance(
   app: FirebaseApp,
   settings?: PerformanceSettings
 ): FirebasePerformance {
+  const provider: Provider<'performance'> = _getProvider(app, 'performance');
+  const perfInstance = provider.getImmediate();
+  perfInstance.init(settings);
```

Comment:
A downside of having extra internal methods on our main controller is that the extra method is just like public ones for js users. Let's add an underscore to give a hint the users should not be calling this mehod.

Reply:
That's why I created another interface. Right now there are 2 different interfaces:

1. FirebasePerformance
2. FirebasePerformanceInternal

`Provider<'performance'>` returns `FirebasePerformanceInternal`, which has the extra methods (so we have access to them here). But users don't use this provider; they use this `getPerformance` method, which returns `FirebasePerformance` (instead of `FirebasePerformanceInternal`), so they don't have access to that method, since it's not defined in `FirebasePerformance` (users would have to do a hard-cast from `FirebasePerformance` to `FirebasePerformanceInternal` - but if they do, I don't think an underscore would prevent them from accessing the method; the only reason to do a hard-cast would be to access that method).

Let me know what you think.

Reply:
Thanks, I forgot that the types get stripped :(
Added the underscore.